gazelle/fileinfo_test.go

```diff
@@ -0,0 +1,88 @@
+/* Copyright 2018 The Bazel Authors. All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+   http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package gazelle
+
+import (
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"reflect"
+	"testing"
+)
+
+func TestProtoFileInfo(t *testing.T) {
```

Comment:
Add comments for what this is and how it's used?

Reply:
The `Proto` part of this function was never supposed to be there. It was copy-pastad. In golang your tests are put in files that end in `_test.go`. Any function that starts with `Test` is called and passed an instance of the `testing.T` object. Idiomatically, this function just tests FileInfo. `Test*` methods that live in `_test.go` files are not exported as a part of the public interface. I've additionally included a comment to indicate what this is testing.